DLImageLoader/DLILOperation.swift

```diff
@@ -49,31 +49,31 @@ class DLILOperation: NSOperation, NSURLConnectionDelegate {
         } else {
             self.completed = completed
             self.canceled = canceled
-            self.connection = NSURLConnection(request: request, delegate: self)
+            let connection = NSURLSession(configuration: NSURLSessionConfiguration.defaultSessionConfiguration(), delegate: self, delegateQueue: nil)
+            self.connection = connection.dataTaskWithRequest(request)
+            self.connection.resume()
             if self.connection == nil {
                 cancel()
             }
         }
     }
     
     /**
-     * pragma mark - NSURLConnectionDataDelegate
+     * pragma mark - NSURLSessionDataDelegate
      */
-    
-    func connection(connection: NSURLConnection!, didReceiveData data: NSData!)
+
+    func URLSession(session: NSURLSession, dataTask: NSURLSessionDataTask, didReceiveData data: NSData)
     {
         self.data.appendData(data)
     }
-    
-    func connectionDidFinishLoading(connection: NSURLConnection!)
+
+    func URLSession(session: NSURLSession, task: NSURLSessionTask, didCompleteWithError error: NSError?)
     {
+        // fail loading
+        if (error != nil) {
+            self.completed?(error: nil, image: nil)
+        }
         // successfull loading
         self.completed?(error: nil, image: UIImage(data: self.data))
```

Comment:
Do I have to embed successful handler in an else block?

`} else { 
self.completed?(error: nil, image: UIImage(data: self.data))
 }`